cps/src/main/java/org/jenkinsci/plugins/workflow/cps/EnvActionImpl.java

```diff
@@ -108,4 +109,23 @@
         owner = r;
     }
 
+    @Extension public static class Binder extends Singleton {
+        @Override public String getName() {
+            return "env";
+        }
+        @Override public Object getProperty(CpsScript script) throws Exception {
+            Run<?,?> run = script.$build();
+            if (run != null) {
+                EnvActionImpl action = run.getAction(EnvActionImpl.class);
+                if (action == null) {
+                    action = new EnvActionImpl();
+                    run.addAction(action);
+                }
+                return action;
```

Comment:
My point was ... you had to jump through ugly and unintuitive code hoops in this PR, just to get the `EnvVars` for a `Run`. Seems like an obvious addition to the `Run` API.